### PR TITLE
Sns interface config

### PR DESCRIPTION
**Pull request by gregorydayao12**
## Description
Sensors module reconfigured to use new utilities. Old code cleaned up to remove last-minute additions from California.

## Changes
- SSR scheme for batteries moved to gpiomanager for easy addition of digital output devices
- Constants added to config

## Additional Info 
screenshots, commands to run etc.

### CODE REVIEW

File: src/utils/config.hpp
Comment (mkristien) [inline src/utils/config.hpp]: What naming scheme is this following? Should not be `int master;`?

File: src/data/data.hpp
Comment (mkristien) [inline src/data/data.hpp]: Please, remove these completely. Or is this something to be done later? At the moment, it seems config stuff is spread all over the place, some in `data` some in `config`. Could you move it all to `config`?

File: src/sensors/gpio_manager.hpp
Comment (mkristien) [inline src/sensors/gpio_manager.hpp:50]: Please add docs. I have no idea what `GpioManager` is supposed to do, hence cannot review. I know lots of places lack documentation but let us try have at least some decent docs for new stuff we are adding. Even better if you can add some unit tests for the new stuff :)

**Review by mkristien — CHANGES_REQUESTED** [review]

File: src/data/data.hpp
Comment (gregorydayao12) [reply src/data/data.hpp]: I did not know whether we are going to do everything from config or data. Won't be my decision/responsibility to take it out of data.hpp without approval, that's why I just added a comment. Do you reckon take it out?

Comment (mkristien) [reply src/data/data.hpp]: My suggestion is to put everything that can change during testing, development, or in Spacex parking lot, e.g. number of IMUs, into config. Double check with heads if uncertain.

File: src/sensors/gpio_manager.hpp
Comment (gregorydayao12) [reply src/sensors/gpio_manager.hpp:50]: Added a [wiki](https://github.com/Hyp-ed/hyped-2020/wiki/GpioManager:-adding-digital-output-devices) for GpioManager that explains functionality if other submodules need to use similar digital output devices (ex embrakes).

File: src/data/data.hpp
Comment (gregorydayao12) [reply src/data/data.hpp]: Going to talk with state machine in regards to these, waiting for their response to see how it is best to approach this, but those changes should be independent of this pr since its a bigger change in data.hpp.

File: src/sensors/gpio_manager.hpp
Comment (mkristien) [reply src/sensors/gpio_manager.hpp:50]: Can you add a link here for quick reference? I also usually appreciate a short description directly in the code at the point of declaration. Thanks.

File: src/sensors/manager_interface.hpp
Comment (mkristien) [inline src/sensors/manager_interface.hpp]: Why do you have these different interfaces that do exactly the same thing? Will you have different implementation of e.g. BmsManagerInterface?

File: src/utils/config.cpp
Comment (mkristien) [inline src/utils/config.cpp]:
```suggestion
        sensors.chip_select.push_back(atoi(value));
```

File: src/utils/config.hpp
Comment (mkristien) [inline src/utils/config.hpp]: I am not sure about the naming scheme here. This is technically not a constant but should behave like one once initialised. I am actually ok with the naming like this but might be worth checking with the heads.

File: src/utils/config.hpp
Comment (mkristien) [inline src/utils/config.hpp]: Please, change name to something more descriptive or add a comment saying what this is.

File: src/sensors/gpio_manager.hpp
Comment (mkristien) [inline src/sensors/gpio_manager.hpp]: Style guide dictates functions should be declared before member variables.

File: src/sensors/gpio_manager.hpp
Comment (mkristien) [inline src/sensors/gpio_manager.hpp]: TODO comment is useless. No idea what `to change`.

File: src/sensors/gpio_manager.hpp
Comment (mkristien) [inline src/sensors/gpio_manager.hpp]: These comments seem wrong. The description implies a functionality not a piece of data.

**Review by mkristien — DISMISSED** [review]

File: src/sensors/manager_interface.hpp
Comment (gregorydayao12) [reply src/sensors/manager_interface.hpp]: Come to think of it, ManagerInterface is quite pointless...

**Review by videosquared — APPROVED** [review]
All changed looks correct and nothing wrong stands out to me

**Pull request merged**
